### PR TITLE
[FEAT] `argilla server`: add user and server id on telemetry metrics

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR restores the server_id for telemetry purposes and also add the user.id and user.role when tracking API requests. 

**Type of change**

- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/src/argilla_server/telemetry.py [hunk 1]
```diff
@@ -16,26 +16,31 @@
...
     server_deployment_type,
 )
+from argilla_server.security.authentication.provider import get_request_user
 
 _LOGGER = logging.getLogger(__name__)
 
 
 @dataclasses.dataclass
 class TelemetryClient:
+    _server_id: int = uuid.getnode()
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/telemetry.py]: The `uuid.getnode` returns an integer value. We can use it "as is".

File: argilla-server/src/argilla_server/telemetry.py [hunk 2]
```diff
@@ -47,16 +52,11 @@ def __post_init__(self):
...
-        library_name = "argilla/server"
-        topic = f"{library_name}/{topic}"
-
-        user_agent = {**data}
-        if include_system_info:
-            user_agent.update(self._system_info)
-        if count is not None:
-            user_agent["count"] = count
+    async def track_data(self, topic: str, data: dict):
+        library_name = "argilla-server"
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/telemetry.py:57]: This aligns the library name (argilla-server) to the package published in pypi.org

File: argilla-server/src/argilla_server/telemetry.py (hunk 1 above)
Comment (jfcalvo) [reply argilla-server/src/argilla_server/telemetry.py]: We should check that `uuid.getnode()` is returning the same value on HF Spaces even after a factory rebuild.

**Review by jfcalvo — APPROVED** [review]

**Review by jfcalvo — APPROVED** [review]

**Comment by frascuchon** [comment]
> We should check that uuid.getnode() is returning the same value on HF Spaces even after a factory rebuild.

Changes introduced storing the server ID allow reuse of the server ID on restart when persistent storage is enabled. We can add also info about persistent storage on server startup.

**Pull request merged**